### PR TITLE
Add Mock.ClearInvocations and Mock.Reset per-object overloads

### PR DESCRIPTION
**Pull request by MladenStankov**
#
This pull request introduces new functionality to allow clearing invocation history and resetting arrangements on mock objects, along with comprehensive tests for these features. The main changes include the implementation of `Mock.ClearInvocations` and `Mock.Reset` methods, updates to the internal repository to support these operations, and a new test fixture to verify their behavior.

### New Mock Control Features

* Added `Mock.ClearInvocations(object instance)` to clear invocation history for a mock without removing its arrangements. This lets you reset the call counters so that only subsequent invocations are counted in assertions. 
* Added `Mock.Reset(object instance)` to remove all arrangements and clear invocations for a mock, while keeping the mock usable for new arrangements.

### Internal Infrastructure

* Implemented `ClearInvocations` and `ResetInstance` methods in `MocksRepository`, handling validation, counter resets, and arrangement removal for mock objects.
* Added a `Reset` method to `InvocationOccurrenceBehavior` to reset the call count for arrangements.

### Testing

* Introduced a new test fixture `MockClearInvocationsFixture.cs` that thoroughly tests the behavior of clearing invocations and resetting mocks, including edge cases and error handling.

These changes improve the flexibility and testability of mock objects, allowing for more precise control over invocation tracking and arrangement management.

closes [#JM-201](https://progresssoftware.atlassian.net/jira/software/c/projects/JM/boards/10340?selectedIssue=JM-201)

### CODE REVIEW

File: Telerik.JustMock/Core/MocksRepository.cs [hunk 1]
```diff
@@ -899,6 +909,72 @@ internal void AssertAll(string message)
...
+
+            MockingUtil.UnwrapDelegateTarget(ref instance);
+            if (instance == null)
+                throw new ArgumentException("Object is not a JustMock mock instance.", "instance");
+
+            var methodMocks = GetMethodMocksFromObject(instance);
+            if (methodMocks.Count == 0 && GetMockMixin(instance, null) == null)
+                throw new ArgumentException("Object is not a JustMock mock instance.", "instance");
+
+            this.DeferInOrderViolations = true;
```
Comment (ValentinaPetrova) [inline Telerik.JustMock/Core/MocksRepository.cs]: DeferInOrderViolations is set to true here, but I don't see it restored afterward. That looks like it can change InOrder behavior for the rest of the repository and defer ordering failures until Mock.Assert instead of failing at the call site.

**Review by ValentinaPetrova — APPROVED** [review]

**Review by ivo-stoilov — APPROVED** [review]
